### PR TITLE
Use nn.softmax in keras.activations.softmax

### PR DESCRIPTION
**Pull request by szutenberg**
There are two issues with [tf.keras.activations.softmax](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/activations.py#L47-L84 ) :

**First**: 
it implements softmax for 3D+ tensors in not optimal way due to old workarounds:
- **Keras**:   Support for 3D+ tensors was added on Nov 29, 2015 [https://github.com/keras-team/keras/commit/8f2d6d2714aa1b60950a2fc355d39297b7f2cdfb](https://github.com/keras-team/keras/commit/8f2d6d2714aa1b60950a2fc355d39297b7f2cdfb)
- **TF**: In that time tensorflow did not support multiple dimensions: issue was reported on Jun 29, 2016 (Softmax for multiple dimensions) [https://github.com/tensorflow/tensorflow/issues/3101](https://github.com/tensorflow/tensorflow/issues/3101)
- **TF**: Support for multiple dimensions was added on Aug 29, 2016 [https://github.com/tensorflow/tensorflow/commit/aeac274ed160618afa242512c38d1fd496466136](https://github.com/tensorflow/tensorflow/commit/aeac274ed160618afa242512c38d1fd496466136)
- **TF**: Improved kernel accepting shapes with higher ranks was added on Aug 6, 2018 [https://github.com/tensorflow/tensorflow/commit/f3aa5d4717deac34b00979fdc4f13c3c9170e819](https://github.com/tensorflow/tensorflow/commit/f3aa5d4717deac34b00979fdc4f13c3c9170e819)

For 3D tensor, the softmax gradient is getting really complicated which seriously hurts performance. example:
image

**Second**:
It returns wrong results for 2D tensors, axis=0. I also prepared unit test for this scenario.

### CODE REVIEW

**Comment by szutenberg** [comment]
@googlebot I signed it!

**Review by fchollet — APPROVED** [review]
LGTM, thanks

**Review by gbaned — APPROVED** [review]
